Functions/ImageProcessing/xASL_im_MergePWI4D.m

```diff
@@ -0,0 +1,101 @@
+function [path_PWI4D, path_PWI4D_Pop] = xASL_im_MergePWI4D(x)
+% xASL_im_MergePWI4D merges several PWIs based on the list in x.modules.asl.sessionsToMerge
+%
+% FORMAT: [path_PWI4D, path_PWI4D_Pop] = xASL_im_MergePWI4D(x)
+%
+% INPUT:
+%   x                   - structure containing fields with all information required to run this submodule (REQUIRED)
+%
+% OUTPUT: 
+%   path_PWI4D          - Path to the new merged file
+%   path_PWI4D_Pop      - Path to the new merged file in the Population directory
+% OUTPUT FILES: 
+%   NIfTI files (see paths above) containing the merged sessions.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function takes previously created PWI4D.nii from sessions described in the x.modules.asl.sessionsToMerge and 
+%              merges them to a single NII with a predefined name. It returns this new name for both native and standard space and it also creates the 
+%              appropriate JSON sidecars.
+%
+% EXAMPLE: 
+%   [path_PWI4D, path_PWI4D_Pop] = xASL_im_MergePWI4D(x)
+% __________________________________
+% Copyright (C) 2015-2024 ExploreASL
+
+% 0 Admin
+if nargin<1 || isempty(x)
+	error('Empty input parameters');
+end
+
+fprintf('Concatenating PWI4D for sessions:');
+% Do this for native and standard space
+pathPWI4D = {x.P.Path_PWI4D x.P.Pop_Path_PWI4D};
+
+% Name of the merged session
+sessionName = [];
+
+for iSpace = 1:2
+	for iSession = 1:numel(x.modules.asl.sessionsToMerge)
+		% Here we get the path of the session, by replacing the session names 
+		pathCurrentPWI4D = replace(pathPWI4D{iSpace}, x.modules.asl.sessionsToMerge{end}, x.modules.asl.sessionsToMerge{iSession});
+
+		% Load NII and JSON
+		if xASL_exist(pathCurrentPWI4D, 'file')
+			[imPWI4Dcurrent, jsonPWI4Dcurrent] = xASL_io_Nifti2Im(pathCurrentPWI4D); % Don't even bother converting to Legacy before merging
+		else
+			% If one of the sessions are missing, then we issue an error
+			error(['Cannot concatenate sessions for subject ' x.SUBJECT ': session '  x.modules.asl.sessionsToMerge{iSession} ' is missing.']);
+		end
+
+		% Error with a missing JSON
+		if isempty(jsonPWI4Dcurrent)
+			error(['Cannot concatenate sessions for subject ' x.SUBJECT ': session '  x.modules.asl.sessionsToMerge{iSession} ' has missing JSON information.']);
```

Comment:
if we merge and don't have JSONS, then we are fucked later on :)

I'm fine doing that, but we should maybe release version 2.0 and keep only a reasonable compatibility... If you process part of ASL module with version 1.11.0 and second part of ASL module with 1.12.0, it will crash. But I thought that we agreed upon previoulsy that nobody should do a dirty thing like this :)

Reply:
Ok, then only:

"has missing JSON information" -> is too general, information about a JSON can be anything.
I would say "does not have a JSON sidecar"